core/src/toga/documents.py

```diff
@@ -92,17 +92,18 @@ def app(self) -> App:
         return self._app
 
     @property
-    def main_window(self) -> Window:
+    def main_window(self) -> Window | None:
         """The main window for the document."""
         return self._main_window
 
     @main_window.setter
-    def main_window(self, window):
+    def main_window(self, window: Window) -> None:
         self._main_window = window
 
     def show(self) -> None:
         """Show the :any:`main_window` for this document."""
-        self.main_window.show()
+        # TODO:PR: does this need a None check for main_window?
+        self.main_window.show()  # type: ignore[union-attr]
```

Comment:
Again this will be a reworked a bit by #2244; but in the case of Document, at the point that show() is being invoked, it should have a main window. If it doesn't, that's an error in the Document class itself.